tests/test_ext_commands.py

```diff
@@ -134,6 +137,24 @@ def test_sanitize_repo_commits_non_processed_files_to_target_branch(
     )
 
 
+def test_sanitize_repo_returns_fail_when_repo_has_staged_changes(
+    sanitizer_config, fake_repo
+):
+    tracked_file = fake_repo.file_infos[0].abspath
+    tracked_file.write_text("this is  the new text!")
+    fake_repo.repo.git.add(tracked_file)
+
+    sanitize_repo = sanitizer.SanitizeRepo()
+    cmd = sanitize_repo.create_extension_command()
+    args = cmd.parser.parse_args(
+        f"--file-list {fake_repo.file_list_path} --repo-root {fake_repo.path}"
+        " --no-commit".split()
+    )
+    result = cmd.callback(args, None)
```

Comment:
Extract this into a helper function called something like `execute_sanitize_repo(args: str)`

```suggestion
    result = execute_sanitize_repo(f"--file-list {fake_repo.file_list_path} ....")
```